extension-androidauto/src/main/java/com/mapbox/maps/extension/androidauto/MapboxCarMap.kt

```diff
@@ -99,4 +99,12 @@ class MapboxCarMap(
   fun clearObservers() {
     carMapSurfaceOwner.clearObservers()
   }
+
+  /**
+   * Override the car gestures with your own implementation. You can always reset it to the default
+   * by setting to the default [MapboxCarMapGestures] instance.
+   */
+  fun setGestures(mapGestures: MapboxCarMapGestures) {
```

Comment:
Exposing MapboxCarMapGestures as a open public class is a way to do it, another approach could be providing a setGestuersListener(), and the GestureListener could be something like
```
interface MapboxCarGesturesListener {
  fun onScroll(
    mapboxCarMapSurface: MapboxCarMapSurface,
    visibleCenter: ScreenCoordinate,
    distanceX: Float,
    distanceY: Float
  ) : Boolean
  
  fun onFling(
    mapboxCarMapSurface: MapboxCarMapSurface,
    velocityX: Float,
    velocityY: Float
  ): Boolean
  
  fun onScale(
    mapboxCarMapSurface: MapboxCarMapSurface,
    focusX: Float,
    focusY: Float,
    scaleFactor: Float
  ): Boolean
}
```

Where when returning the true, it suggests the gesture is already handled, and we don't process it further internally?

I feel the latter seems more natural to use and without exposing our MapboxCarMapGestures logic, wdyt?

Reply:
Yeah we can do it this way. Also think we can have a default implementation return false on each of them, that way you are not required to implement them all, unless you're coming from java. 

It's a little more wiring up, but will check it out. 👍

Reply:
It's updated, there were pros and cons. I'm impartial here, potentially biased toward the open class because all the logic is in a single place. 

I haven't squashed the commit, in case we want to go back. But the tests were also improved, so please take a look and let me know what you think.